misc/config_tools/scenario_config/scenario_cfg_gen.py

```diff
@@ -91,35 +91,43 @@ def get_scenario_item_values(board_info, scenario_info):
     return scenario_item_values
 
 
-def validate_scenario_setting(board_info, scenario_info):
+def validate_scenario_schema(scenario_info):
     """
     Validate settings in scenario xml if there is scenario schema
     :param xsd_doc: scenario schema
     :param scenario_info: scenario file
     """
     try:
         import xmlschema
+    except ImportError:
+        return
+
+    """
+    XMLSchema does not process XInclude.
+    Use lxml to expand the schema which is feed to XMLSchema as a string.
+    """
+    xsd_doc = etree.parse(common.SCENARIO_SCHEMA_FILE)
+    xsd_doc.xinclude()
+    my_schema = xmlschema.XMLSchema11(etree.tostring(xsd_doc, encoding="unicode"))
 
-        # XMLSchema does not process XInclude. Use lxml to expand the schema which is feed to XMLSchema as a string.
-        xsd_doc = SCENARIO_SCHEMA_FILE
-        xsd_doc.xinclude()
-        my_schema = xmlschema.XMLSchema11(etree.tostring(xsd_doc, encoding="unicode"))
+    it = my_schema.iter_errors(scenario_info)
+    for idx, validation_error in enumerate(it, start=1):
+        key = "idx:path={},{}".format(idx, validation_error.path)
```

Comment:
@yuchuyang Can we make the format of the key in ERROR_LIST same as the legacy format as below?
		{"vm:id=0,name": "VM name length should be in range [1,32] bytes"}
		{"vm:id=0,communication_vuart:id=1,target_vm_id": "target vm 2 disables legacy vuart 1 and all communication vuarts"}

Reply:
@szhen11  updated.
Just keep in mind, this "target vm 2 disables legacy vuart 1 and all communication vuarts" haven't been implemented yet.